var/spack/repos/builtin/packages/mapl/package.py

```diff
@@ -415,4 +420,8 @@ def check(self):
         with working_dir(self.builder.build_directory):
             # The test suite contains a lot of tests. We select only those
             # that are cheap. Note this requires MPI and 6 processes
-            ctest("--output-on-failure", "-L", "ESSENTIAL")
+            args = ["--output-on-failure", "--label-regex", "ESSENTIAL"]
+            if not self.spec.satisfies("skip_tests=none"):
```

Comment:
Can't we use `self.run_tests` here? That will be set to true only if build-time tests have been required
```suggestion
            if not self.run_tests:
```

Reply:
Not really, because all of this is only for when build-time tests are being run anyway. The key here is that for this and eventually other packages, we want/need to be able to specify tests to skip-- sometimes because they break on a specific system, sometimes because the resources needed aren't available on a given system (e.g., large MPI-based tests), and so on. I opted for using a variant because having a working unit testing apparatus is important for us so I felt it was worth it to add a variant even though it affects the hash, _but_ the ideal solution would be some way to specify tests to include/exclude in a way that doesn't affect the package hash (like through an environment variable, or if Spack were to somehow implement test-only variants analogous to test-only dependencies).

Reply:
Yeah, in that case an environment variable would be good - but we'd need support from the package to do it properly.